api/src/main/java/io/opencensus/trace/BaseMessageEvent.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2018, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.trace;
+
+/**
+ * Superclass for {@link MessageEvent} and {@link NetworkEvent} to resolve API backward
```

Comment:
I understand why we are doing this, because eventhough the version is 0.12, we are treating it as a major release. That said, `BaseMessageEvent` seems a bit crap to keep forever. Why was `Event` or similar not ok?

Reply:
The naming itself did seem crap. I didn't use `Event` since it might be misleading to refer to anything happened in the system.
We don't want this class forever, we plan to remove `NetworkEvent` and `BaseMessageEvent` in the future.

Reply:
since we are exposing BaseMessageEvent public, even if hiding ctor, if it is temporary, mark it deprecated. that way no one in their right mind will start casting to it.